sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -427,8 +427,12 @@ case class MapObjects private(
       case _ => ""
     }
 
+    val inputDataType = inputData.dataType match {
+      case p: PythonUserDefinedType => p.sqlType
```

Comment:
Is it possible to catch the python udt before passing it to `MapObjects`? I'm kind of worried about leaking python udt to a lot of places, we should handle them just in a few places.